torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -0,0 +1,215 @@
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/freeze_module.h>
+
+#include <torch/csrc/jit/graph_executor_impl.h>
+#include <torch/csrc/jit/passes/inliner.h>
+
+#include <stack>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+class AttributePropagator {
+ private:
+  // Contains attributes that can't be folded or user directs to keep them.
+  std::map<script::Module*, std::set<std::string>> preservedAttrs_;
```

Comment:
maybe use `script::MoudlePtr`? also you can get this from `module._ivalue()`

Reply:
`ModulePtr` is a pointer to the real object and that's what should be used as a module "identifier". There could be several `Module` instances referencing the same `ModulePtr`: `Module` is a just thin wrapper that can be copied a lot, its address shouldn't be used as a key in the map.

Reply:
Ok! I saw this type. I suspected to be the safe unique id. Will make the changes. Thank you